packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,613 @@
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('AnotherChildStack');
+    }).toThrow(/Nested Stack 'AnotherChildStack' was not included in the nestedStacks property when including the parent template/);
+  });
+
+  test('correctly handles renaming of references across nested stacks', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('cross-stack-refs.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+    const cfnBucket = parentTemplate.getResource('Bucket');
+    cfnBucket.overrideLogicalId('DifferentBucket');
+    const parameter = parentTemplate.getParameter('Param');
+    parameter.overrideLogicalId('DifferentParameter');
+
+    expect(stack).toHaveResourceLike('AWS::CloudFormation::Stack', {
+      "Parameters": {
+        "Param1": {
+          "Ref": "DifferentParameter",
+        },
+        "Param2": {
+          "Fn::GetAtt": ["DifferentBucket", "Arn"],
+        },
+      },
+    });
+  });
+
+  test('returns the CfnStack object from getResource() for a nested stack that was not in the nestedStacks property', () => {
+    const cfnTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-two-children.json'),
+    });
+
+    const childStack1 = cfnTemplate.getResource('ChildStack');
+
+    expect(childStack1).toBeInstanceOf(core.CfnStack);
+  });
+
+  test('returns the CfnStack object from getResource() for a nested stack that was in the nestedStacks property', () => {
+    const cfnTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-one-child.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    const childStack1 = cfnTemplate.getResource('ChildStack');
+
+    expect(childStack1).toBeInstanceOf(core.CfnStack);
+  });
+
+  test('attributes are correctly parsed', () => {
+    new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-with-attributes.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+        'AnotherChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CloudFormation::Stack', {
+      "Metadata": {
+        "Property1": "Value1",
+      },
+      "DeletionPolicy": "Retain",
+      "DependsOn": [
+        "AnotherChildStack",
+      ],
+      "UpdateReplacePolicy": "Retain",
+    }, ResourcePart.CompleteDefinition);
+  });
+
+  test('stacks created by the finder are correctly nested', () => {
```

Comment:
1. This is a copy&paste from the above code, minus the last line. I would just remove the test above.
2. The description here is baaaad 🙂. `finder` is an implementation detail. It has no place being mentioned in the test's description.

I would just re-use the description from the above test here:

```suggestion
  test('handles Metadata, DeletionPolicy and UpdateReplacePolicy attributes of the nested stack' resource, () => {
```